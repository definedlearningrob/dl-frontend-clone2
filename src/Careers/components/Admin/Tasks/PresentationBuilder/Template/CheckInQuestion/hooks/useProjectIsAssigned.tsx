import { useMemo } from 'react';
import { useParams } from 'react-router-dom';
import { useQuery } from '@apollo/client';
import { isEmpty } from 'lodash-es';

import teamProjectQuery from '@pbl/graphql/student/queries/teamProject';
import projectQuery from '@pbl/graphql/student/queries/project';

import { useRole } from '@shared/hooks/useRole';

export const useProjectIsAssigned = () => {
  const { isStudent } = useRole();
  const { projectId, teamId } = useParams<{ projectId: string; teamId: string }>();

  const properQuery = useMemo(() => (teamId ? teamProjectQuery : projectQuery), [teamId]);

  const queryVariables = useMemo(
    () =>
      teamId
        ? { id: projectId, teamId, track: true, trackPresentation: true }
        : { id: projectId, track: true, trackPresentation: true },
    [teamId]
  );

  const { data, error, loading } = useQuery(properQuery, {
    variables: queryVariables,
    skip: !isStudent,
  });

  if (!isStudent || loading || error) {
    return false;
  }

  return !isEmpty(data?.project?.assignedAt);
};
