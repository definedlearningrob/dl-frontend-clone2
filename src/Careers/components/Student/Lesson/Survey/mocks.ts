import lessonInCourseQuery from '@dc/graphql/student/queries/lessonInCourse';
import { CREATE_CAREER_REVIEW_SURVEY_ATTEMPT } from '@dc/graphql/student/mutations/createCareerReviewSurveyAttempt';
import createCareerReviewSurveyAnswersMutation from '@dc/graphql/student/mutations/createCareerReviewSurveyAnswers';
import currentCoursesQuery from '@dc/graphql/student/queries/currentCourses';
import { SURVEY_QUESTION_TYPE } from '@dc/components/Student/Lesson/Survey/types';

export const createAttemptSpy = jest.fn();
export const lessonInCourseSpy = jest.fn();

export let createCareerReviewSurveyAnswersSpy = jest.fn();
export let currentCoursesSpy = jest.fn();

export const mocks = [
  {
    request: {
      query: lessonInCourseQuery,
      variables: { courseId: '2', lessonId: '111', track: true },
    },
    result: () => {
      lessonInCourseSpy();

      return {
        data: {
          course: {
            careerName: 'Dental Laboratory Technicians',
            content: {
              checkIns: [],
              items: [],
            },
            hasInstitutionsInStudentState: true,
            id: '2',
            lesson: {
              assignments: [],
              attachments: [],
              careerReviewSurvey: {
                version: 1,
                performed: false,
                questions: [
                  {
                    answer: [],
                    id: '1',
                    options: [
                      {
                        option: 'Yes, definitely',
                        step: 1,
                      },
                      {
                        option: 'Definitely not',
                        step: 2,
                      },
                    ],
                    question: 'Can blind people see their dreams?',
                    type: 'single_choice' as SURVEY_QUESTION_TYPE,
                  },
                  {
                    answer: [],
                    id: '2',
                    options: [
                      {
                        option: 'Very Interested',
                        step: 1,
                      },
                      {
                        option: 'Not at all Interested',
                        step: 2,
                      },
                      {
                        option: 'Other',
                        step: 3,
                      },
                    ],
                    question: 'How interested are you in the Pathway this course is a part of?',
                    type: 'multiple_choice' as SURVEY_QUESTION_TYPE,
                  },
                ],
                __typename: 'CareerReviewSurvey',
              },
              checkInGroups: [],
              checkInQuestions: [],
              description: {
                introduction: null,
                goal: null,
                role: null,
                audience: null,
                situation: null,
                __typename: 'LessonDescription',
              },
              hasPresentation: false,
              id: '609',
              imageUrl: 'http://test.com/image.png',
              name: 'Career Review Survey',
              externalPresentations: [],
              researchLinks: [],
              texts: [],
              type: 'career_review_survey',
              videos: [],
              vocabularies: [],
              __typename: 'Lesson',
            },
            lessons: [
              {
                id: '782',
                step: 4,
                careerReviewSurvey: null,
              },
              {
                id: '609',
                step: 5,
                careerReviewSurvey: {
                  performed: false,
                  version: 1,
                  __typename: 'CareerReviewSurvey',
                },
              },
              {
                id: '275',
                step: 1,
                careerReviewSurvey: null,
              },
              {
                id: '331',
                step: 2,
                careerReviewSurvey: null,
              },
              {
                id: '668',
                step: 3,
                careerReviewSurvey: null,
              },
            ],
            name: 'Dental Laboratory Technician',
            progress: {
              submitted: 10,
              total: 10,
            },
            reviewSurvey: {
              questions: [
                {
                  id: '1',
                },
                {
                  id: '2',
                },
                {
                  id: '3',
                },
                {
                  id: '4',
                },
                {
                  id: '5',
                },
                {
                  id: '6',
                },
                {
                  id: '7',
                },
                {
                  id: '8',
                },
                {
                  id: '9',
                },
                {
                  id: '10',
                },
                {
                  id: '11',
                },
              ],
            },
          },
        },
      };
    },
  },
  {
    request: {
      query: CREATE_CAREER_REVIEW_SURVEY_ATTEMPT,
      variables: { input: { contextType: 'COURSE', contextId: '1' } },
    },
    result: () => {
      createAttemptSpy();

      return {
        data: {
          createCareerReviewSurveyAttempt: {
            careerReviewSurveyAttempt: {
              id: '2',
              status: 'IN_PROGRESS',
            },
          },
        },
      };
    },
  },
  {
    request: {
      query: createCareerReviewSurveyAnswersMutation,
      variables: {
        input: {
          contextId: '1',
          contextType: 'COURSE',
          answers: [
            { questionId: '1', answer: ['Yes, definitely'] },
            { questionId: '2', answer: ['Very Interested'] },
          ],
        },
      },
    },
    result: () => {
      createCareerReviewSurveyAnswersSpy();

      return {
        data: {
          createCareerReviewSurveyAnswers: {
            status: 'success',
          },
        },
      };
    },
  },
  {
    request: {
      query: currentCoursesQuery,
    },
    result: () => {
      currentCoursesSpy();

      return {
        data: {
          currentCourses: {
            id: '1',
            imageUrl: '',
            name: '',
            progress: {
              submitted: 2,
              total: 5,
            },
            status: 'success',
            pathway: {
              name: '',
            },
            thumbnailUrl: '',
            type: 'survey',
          },
        },
      };
    },
  },
];
