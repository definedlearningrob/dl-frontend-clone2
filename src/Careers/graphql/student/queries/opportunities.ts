import { gql } from '@apollo/client';

import { OPPORTUNITY_TYPE } from '@dc/resources/enums';
import {
  TOpportunity as TBaseOpportunity,
  TVirtualInternship as TBaseVirtualInternship,
} from '@dc/resources/types';

export default gql`
  query StudentOpportunities($page: Int, $perPage: Int, $filter: StudentOpportunityFilter) {
    opportunities(page: $page, perPage: $perPage, filter: $filter) {
      nodesCount
      pagesCount
      nodes {
        id
        name
        applicationStatus
        isFavorite
        isRecommended
        imageUrl
        opportunityType
        deadline
        periodStart
        partner {
          id
          name
        }
        periodEnd
        virtualInternship {
          id
          requiredExperiences
          roadmapItemsCount
          readinessSkillsLessons {
            id
          }
          status
        }
        pathways {
          id
          name
        }
        imageFitToContainer
      }
    }
  }
`;

export type TOpportunitiesVariables = {
  page: number;
  perPage: number;
  filter?: TOpportunitiesFilter;
};

export type TOpportunitiesData = {
  opportunities: {
    nodesCount: number;
    pagesCount: number;
    nodes: TOpportunity[];
  };
};

type TPathway = Pick<TBaseOpportunity['pathways'][number], 'id' | 'name'>;

type TVirtualInternship = Pick<
  TBaseVirtualInternship,
  'id' | 'requiredExperiences' | 'roadmapItemsCount' | 'status'
> & {
  readinessSkillsLessons: {
    id: string;
  }[];
};

export type TOpportunity = Pick<
  TBaseOpportunity,
  | 'id'
  | 'name'
  | 'applicationStatus'
  | 'isFavorite'
  | 'isRecommended'
  | 'imageUrl'
  | 'opportunityType'
  | 'deadline'
  | 'periodStart'
  | 'periodEnd'
  | 'partner'
  | 'imageFitToContainer'
> & {
  pathways: TPathway[];
  virtualInternship: TVirtualInternship | null;
};

type TOpportunitiesFilter = {
  nameCont?: string;
  pathwaysIdIn?: string[];
  tagsContain?: string[];
  typeEq?: OPPORTUNITY_TYPE;
};
