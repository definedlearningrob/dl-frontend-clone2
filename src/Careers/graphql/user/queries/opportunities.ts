import { gql, TypedDocumentNode } from '@apollo/client';
import { OpportunityTypes } from '@graphql/dc/shared/types';
import { Opportunity, Partner } from '@graphql/dc/users/types';

import { VISIBILITY_SCOPE } from '@dc/resources/enums';
import { TOpportunity as TBaseOpportunity } from '@dc/resources/types';

export const OPPORTUNITIES_QUERY: TypedDocumentNode<
  TOpportunitiesData,
  TOpportunitiesVariables
> = gql`
  query UserOpportunities($page: Int, $perPage: Int, $filter: OpportunityFilter) {
    opportunities(page: $page, perPage: $perPage, filter: $filter) {
      nodesCount
      pagesCount
      nodes {
        id
        createdAt
        name
        opportunityType
        periodEnd
        periodStart
        deadline
        visibilityScope
        hasPendingApplications
        imageUrl
        thumbnailUrl
        entities {
          uuid
        }
        partner {
          id
          name
        }
        pathways {
          name
        }
        imageFitToContainer
      }
    }
  }
`;

export type TOpportunitiesVariables = {
  page?: number;
  perPage?: number;
  filter?: TOpportunitiesFilter;
};

export type TOpportunitiesData = {
  opportunities: {
    nodesCount: number;
    pagesCount: number;
    nodes: (Pick<
      TOpportunity,
      | 'id'
      | 'name'
      | 'opportunityType'
      | 'periodEnd'
      | 'periodStart'
      | 'deadline'
      | 'visibilityScope'
      | 'createdAt'
      | 'hasPendingApplications'
      | 'imageUrl'
      | 'thumbnailUrl'
      | 'virtualInternship'
      | 'imageFitToContainer'
    > & {
      entities: Pick<Opportunity['entities'][number], 'uuid'>[];
      pathways: Pick<TBaseOpportunity['pathways'][number], 'name'>[];
      partner: Pick<Partner, 'id' | 'name'> | null;
    })[];
  };
};

type TOpportunitiesFilter = {
  nameCont?: string;
  pathwaysIdIn?: string[];
  tagsContain?: string[];
  typeEq?: OpportunityTypes;
};

export const OPPORTUNITY_QUERY: TypedDocumentNode<TOpportunityData, TOpportunityVariables> = gql`
  query UserOpportunity(
    $id: ID!
    $page: Int
    $perPage: Int
    $filter: OpportunityApplicationFilter
    $track: Boolean
  ) {
    opportunity(id: $id, track: $track) {
      id
      name
      automaticAcceptance
      availableSpots
      creditsOutcomes
      description
      imageUrl
      location
      opportunityType
      pathways {
        id
        name
      }
      salaryInformation
      tags
      deadline
      periodEnd
      periodStart
      applications(page: $page, perPage: $perPage, filter: $filter) {
        nodes {
          id
          appliedAt
          updatedAt
          status
          student {
            uuid
            fullName
            schoolClasses {
              uuid
              name
            }
          }
        }
      }
      entities {
        uuid
        name
      }
      visibilityScope
      hasPendingApplications
      virtualInternship {
        id
      }
      partner {
        id
        name
      }
      imageFitToContainer
      thumbnailUrl
    }
  }
`;

export type TOpportunityData = {
  opportunity: TOpportunity;
};

export type TOpportunityApplicationFilter = {
  studentFullNameCont: string;
};

type TOpportunityVariables = {
  id: string;
  perPage?: number;
  page?: number;
  filter?: TOpportunityApplicationFilter;
  track: boolean;
};

export const CREATE_OPPORTUNITY_MUTATION: TypedDocumentNode<
  TCreateOpportunityData,
  TCreateOpportunityVariables
> = gql`
  mutation CreateOpportunity($input: CreateOpportunityMutationInput!) {
    createOpportunity(input: $input) {
      opportunity {
        description
        name
        opportunityType
        pathways {
          id
          name
        }
        automaticAcceptance
        availableSpots
        creditsOutcomes
        entities {
          uuid
          name
        }
        visibilityScope
        imageUrl
        location
        tags
        salaryInformation
        periodStart
        periodEnd
        deadline
        id
        partner {
          id
          name
        }
      }
    }
  }
`;

type TCreateOpportunityData = {
  createOpportunity: {
    opportunity: TSaveOpportunity;
  };
};

type TCreateOpportunityVariables = {
  input: {
    imageFilename?: string | null;
    imageUuid?: string | null;
    description: string;
    name: string;
    opportunityType: OpportunityTypes;
    pathwayIds: string[];
    automaticAcceptance: boolean;
    availableSpots: number;
    creditsOutcomes: string;
    location: string | null;
    tags: string[];
    salaryInformation: string | null;
    periodStart: Date | null;
    periodEnd: Date | null;
    deadline: Date | null;
    visibilityScope: VISIBILITY_SCOPE;
    entityUuids: string[];
    partnerId: string | null;
  };
};

export const UPDATE_OPPORTUNITY_MUTATION: TypedDocumentNode<
  TUpdateOpportunityData,
  TUpdateOpportunityVariables
> = gql`
  mutation UpdateOpportunity($input: UpdateOpportunityMutationInput!) {
    updateOpportunity(input: $input) {
      opportunity {
        id
        name
        automaticAcceptance
        availableSpots
        creditsOutcomes
        description
        imageUrl
        location
        opportunityType
        pathways {
          id
          name
        }
        entities {
          uuid
          name
        }
        salaryInformation
        tags
        deadline
        periodEnd
        periodStart
        visibilityScope
        partner {
          id
          name
        }
      }
    }
  }
`;

type TUpdateOpportunityData = {
  updateOpportunity: {
    opportunity: TSaveOpportunity;
  };
};

type TUpdateOpportunityVariables = {
  input: {
    id: string;
    description: string;
    name: string;
    opportunityType: OpportunityTypes;
    pathwayIds: string[];
    automaticAcceptance: boolean;
    availableSpots: number;
    creditsOutcomes: string;
    imageFilename?: string;
    imageUuid?: string;
    location: string | null;
    tags: string[];
    salaryInformation: string | null;
    periodStart: Date | null;
    periodEnd: Date | null;
    deadline: Date | null;
    visibilityScope: VISIBILITY_SCOPE;
    entityUuids: string[];
    partnerId: string | null;
  };
};

export const ARCHIVE_OPPORTUNITY: TypedDocumentNode<
  TArchiveOpportunityData,
  TArchiveOpportunityVariables
> = gql`
  mutation ArchiveOpportunity($input: ArchiveOpportunityMutationInput!) {
    archiveOpportunity(input: $input) {
      opportunity {
        id
      }
    }
  }
`;

type TArchiveOpportunityData = {
  archiveOpportunity: {
    opportunity: { id: string };
  };
};

type TArchiveOpportunityVariables = {
  input: { id: string };
};

type TSaveOpportunity = Pick<
  TBaseOpportunity,
  | 'id'
  | 'name'
  | 'automaticAcceptance'
  | 'entities'
  | 'availableSpots'
  | 'creditsOutcomes'
  | 'description'
  | 'imageUrl'
  | 'thumbnailUrl'
  | 'location'
  | 'opportunityType'
  | 'salaryInformation'
  | 'tags'
  | 'deadline'
  | 'periodEnd'
  | 'periodStart'
  | 'partner'
> & {
  pathways: Pick<TBaseOpportunity['pathways'][number], 'id' | 'name'>[];
};

export type TOpportunity = Pick<
  TBaseOpportunity,
  | 'id'
  | 'createdAt'
  | 'name'
  | 'automaticAcceptance'
  | 'availableSpots'
  | 'creditsOutcomes'
  | 'description'
  | 'imageUrl'
  | 'thumbnailUrl'
  | 'entities'
  | 'location'
  | 'opportunityType'
  | 'salaryInformation'
  | 'tags'
  | 'deadline'
  | 'periodStart'
  | 'periodEnd'
  | 'applications'
  | 'visibilityScope'
  | 'hasPendingApplications'
  | 'partner'
  | 'imageFitToContainer'
> & {
  pathways: Pick<TBaseOpportunity['pathways'][number], 'id' | 'name'>[];
  virtualInternship: { id: string } | null;
};
