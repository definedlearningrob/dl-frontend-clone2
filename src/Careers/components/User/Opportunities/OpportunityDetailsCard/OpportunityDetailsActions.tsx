import { useTranslation } from 'react-i18next';
import { useParams } from 'react-router';
import { useToggle } from 'react-use';

import { TUserInfo } from '@dc/graphql/user/queries/userInfo';
import useUserInfo from '@dc/hooks/useUserInfo';
import { RecommendOpportunityModal } from '@dc/components/User/Opportunities/RecommendOpportunityModal';

import Link from '@shared/components/Link';
import Button from '@shared/components/Button/Button';
import { useBreakpointUp } from '@shared/hooks/useBreakpointUp';

type Props = {
  virtualInternshipId?: string;
};

export const OpportunityDetailsActions = ({ virtualInternshipId }: Props) => {
  const { t } = useTranslation();
  const isFullHD = useBreakpointUp({ breakpoint: 'xxxl' });
  const [isRecommendModalOpen, toggleRecommendModal] = useToggle(false);

  const {
    userInfo: { permissions },
  } = useUserInfo<TUserInfo>();

  const { id } = useParams<{ id: string }>();

  const isVirtualInternship = !!virtualInternshipId;
  const isWBLAdmin = permissions?.wblAdmin;

  const buttonVariant = isWBLAdmin ? 'primary-outlined' : 'primary';
  const buttonSize = isFullHD ? 'md' : 'sm';

  return (
    <>
      <div className='flex gap-xs xxxl:gap-sm'>
        {isWBLAdmin && (
          <Link
            size={buttonSize}
            to={`/opportunities/${id}/manage-applications`}
            variant={isVirtualInternship ? 'secondary' : 'primary'}>
            {t('user.opportunities.applications')}
          </Link>
        )}
        {isVirtualInternship && (
          <Link
            size={buttonSize}
            to={`/opportunities/${id}/virtual-internship/${virtualInternshipId}`}
            variant='primary'>
            {t('user.opportunities.showInternship')}
          </Link>
        )}
        <Button size={buttonSize} variant={buttonVariant} onClick={toggleRecommendModal}>
          {t('user.opportunities.recommend')}
        </Button>
      </div>
      <RecommendOpportunityModal
        isOpen={isRecommendModalOpen}
        opportunityId={id}
        onClose={() => toggleRecommendModal(false)}
      />
    </>
  );
};
