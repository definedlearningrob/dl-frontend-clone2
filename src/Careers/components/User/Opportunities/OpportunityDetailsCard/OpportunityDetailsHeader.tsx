import { useHistory, useParams } from 'react-router-dom';
import { useTranslation } from 'react-i18next';

import { TUserInfo } from '@dc/graphql/user/queries/userInfo';
import useUserInfo from '@dc/hooks/useUserInfo';
import { VISIBILITY_SCOPE } from '@dc/resources/enums';

import { ReactComponent as EditIcon } from '@shared/svg/edit.svg';
import { ReactComponent as ArchiveIcon } from '@shared/svg/delete_outlined.svg';
import { useBreakpointUp } from '@shared/hooks/useBreakpointUp';
import { Tooltip } from '@shared/components/Tooltip';
import { Kicker } from '@shared/components/Kicker';
import { IconButton } from '@shared/components/IconButton/IconButton';

type Props = {
  visibilityScope: string;
  callArchiveModal: () => void;
  hasPendingApplications: boolean;
};

export const OpportunityDetailsHeader = ({
  visibilityScope,
  callArchiveModal,
  hasPendingApplications,
}: Props) => {
  const history = useHistory();
  const isFullHD = useBreakpointUp({ breakpoint: 'xxxl' });
  const { id } = useParams<{ id: string }>();
  const { userInfo, isSystemAdminUser } = useUserInfo<TUserInfo>();
  const isWBLAdmin = userInfo.permissions.wblAdmin;
  const { t } = useTranslation();
  const handleEditClick = () => {
    history.push(`/opportunities/${id}/edit`);
  };

  if (!visibilityScope) {
    return null;
  }

  const canEdit = (visibilityScope === VISIBILITY_SCOPE.ENTITY && isWBLAdmin) || isSystemAdminUser;

  if (!canEdit) {
    return null;
  }

  const iconSize = isFullHD ? 'lg' : 'md';

  return (
    <div className='flex justify-between items-end flex-1 gap-xs'>
      {hasPendingApplications && (
        <Kicker className='!mb-0' size={isFullHD ? 'md' : 'sm'} variant='secondary'>
          {t('opportunityDetails.pendingApplications')}
        </Kicker>
      )}
      <div className='flex flex-1 justify-end gap-xs'>
        <Tooltip message={t('common.actions.edit')}>
          <IconButton
            Icon={EditIcon}
            size={iconSize}
            variant='primary-outlined'
            onClick={handleEditClick}
          />
        </Tooltip>
        <Tooltip message={t('common.actions.archive')}>
          <IconButton
            Icon={ArchiveIcon}
            size={iconSize}
            variant='danger-outlined'
            onClick={callArchiveModal}
          />
        </Tooltip>
      </div>
    </div>
  );
};
