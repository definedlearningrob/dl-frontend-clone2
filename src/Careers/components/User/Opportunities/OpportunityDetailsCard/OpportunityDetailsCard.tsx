import { useParams } from 'react-router-dom';
import { OpportunityTypes } from '@graphql/dc/shared/types';

import { OpportunitySummaryCard } from '@dc/components/OpportunityDetails/OpportunitySummaryCard';
import { useOpportunityQuery } from '@dc/graphql/user/hooks/useOpportunityQuery';
import { ArchiveOpportunityModal } from '@dc/components/User/Opportunities/OpportunitiesList/ArchiveOpportunityModal';

import { useToggle } from '@shared/hooks/useToggle';

import { OpportunityDetailsActions } from './OpportunityDetailsActions';
import { OpportunityDetailsHeader } from './OpportunityDetailsHeader';
import { Skeleton } from './Skeleton';

export const OpportunityDetailsCard = () => {
  const [isModalOpen, toggleIsModalOpen] = useToggle(false);
  const { id } = useParams<{ id: string }>();
  const { data, loading } = useOpportunityQuery({ id });

  if (loading || !data) {
    return <Skeleton />;
  }

  const {
    opportunity: {
      visibilityScope,
      opportunityType,
      name,
      id: opportunityId,
      virtualInternship,
      hasPendingApplications,
    },
  } = data;
  const isVirtualInternship = opportunityType === OpportunityTypes.VIRTUAL_INTERNSHIP;

  return (
    <>
      <OpportunitySummaryCard
        actions={<OpportunityDetailsActions virtualInternshipId={virtualInternship?.id} />}
        header={
          !isVirtualInternship && (
            <OpportunityDetailsHeader
              callArchiveModal={toggleIsModalOpen}
              hasPendingApplications={hasPendingApplications}
              visibilityScope={visibilityScope}
            />
          )
        }
        opportunity={data.opportunity}
      />
      {isModalOpen && (
        <ArchiveOpportunityModal
          handleClose={toggleIsModalOpen}
          opportunityId={opportunityId}
          opportunityName={name}
        />
      )}
    </>
  );
};
