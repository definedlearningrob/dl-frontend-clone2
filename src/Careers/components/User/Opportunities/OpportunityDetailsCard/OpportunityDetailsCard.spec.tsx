import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { Route } from 'react-router-dom';
import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { renderWithRouterAndReduxProvider } from '@dc/utils/test';
import { OPPORTUNITY_QUERY } from '@dc/graphql/user/queries/opportunities';
import { OPPORTUNITY_APPLICATION_STATUS, Roles } from '@dc/resources/enums';
import { UserInfoProvider } from '@dc/hooks/useUserInfo';
import { userInfoMock } from '@dc/tests/mocks/userMocks';

import { OpportunityDetailsCard } from './OpportunityDetailsCard';

const myOpportunityDetails = {
  request: {
    query: OPPORTUNITY_QUERY,
    variables: { id: '6', track: true },
  },
  result: {
    data: {
      opportunity: {
        id: 6,
        entities: [],
        hasPendingApplications: false,
        visibilityScope: 'GLOBAL',
        name: 'Virtula Intern Ship ',
        automaticAcceptance: true,
        salaryInformation: '1000',
        availableSpots: 1,
        creditsOutcomes: '\u003cp\u003esome text\u0026nbsp;\u003c/p\u003e',
        description: '\u003cp\u003esome text\u0026nbsp;\u003c/p\u003e',
        imageUrl: 'http://localstack.lvh.me:4566/dev-bucket/images/opportunities/tree=1',
        thumbnailUrl: 'http://localstack.lvh.me:4566/dev-bucket/images/opportunities/tree=1',
        location: null,
        opportunityType: 'VIRTUAL_INTERNSHIP',
        pathways: [{ id: '8', name: 'A/V Technology \u0026 Film', __typename: 'Pathway' }],
        tags: [],
        deadline: null,
        periodEnd: null,
        periodStart: null,
        virtualInternship: {
          id: '123',
        },
        applications: {
          nodes: [
            {
              id: '1',
              appliedAt: '',
              updatedAt: '',
              status: OPPORTUNITY_APPLICATION_STATUS.ACCEPTED,
              student: {
                uuid: '1',
                fullName: 'Allison Adams',
                schoolClasses: [
                  {
                    uuid: '2',
                    name: 'Physical Education, Group 9A - 101 - P. McCartney',
                  },
                  {
                    uuid: '3',
                    name: 'Drama 9 - 100 - M. Paucek (Section 1)',
                  },
                  {
                    uuid: '4',
                    name: 'Living Environment - 101 - S. Ogden (Section 1)',
                  },
                  {
                    uuid: '5',
                    name: 'Spanish I - 101 - L. Jacobs (Section 2)',
                  },
                  {
                    uuid: '6',
                    name: 'Class 903, Homeroom - 903 - J. Mertz (Section 3)',
                  },
                  {
                    uuid: '7',
                    name: 'Period 3',
                  },
                  {
                    uuid: '8',
                    name: 'U.S. History, Honors - 102 - J. Doyle (Section 4)',
                  },
                  {
                    uuid: '9',
                    name: 'Group Guidance - 905 - J. Cummings (Section 5)',
                  },
                  {
                    uuid: '10',
                    name: 'Statistics',
                  },
                  {
                    uuid: '11',
                    name: 'Ninth Grade English - 101 - B. Glover (Section 1)',
                  },
                  {
                    uuid: '12',
                    name: 'Sculpture - 004 - J. Page (Section 4)',
                  },
                ],
              },
            },
          ],
        },
        partner: null,
        imageFitToContainer: false,
      },
    },
  },
};

const renderComponentCardWithProvider = (mocks: MockedResponse[] = []) =>
  renderWithRouterAndReduxProvider(
    <MockedProvider mocks={mocks}>
      <UserInfoProvider
        value={{ userInfo: { ...userInfoMock.result.data.userInfo, role: Roles.SYSTEM_ADMIN } }}>
        <Route path='/opportunities/:id'>
          <OpportunityDetailsCard />
        </Route>
      </UserInfoProvider>
    </MockedProvider>,
    { route: '/opportunities/6' }
  );

describe('OpportunityDetailsCard', () => {
  it('should render correctly', async () => {
    const { container } = renderComponentCardWithProvider([myOpportunityDetails]);

    expect(await screen.findByRole('heading', { name: 'Virtula Intern Ship' })).toBeInTheDocument();
    expect(await screen.findByText('Virtual Internship')).toBeInTheDocument();
    expect(await screen.findByText('A/V Technology & Film')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('navigates to virtual internship after clicking a button', async () => {
    const { history } = renderComponentCardWithProvider([myOpportunityDetails]);
    history.push = jest.fn();

    const showInternshipButton = await screen.findByRole('button', { name: 'Show internship' });
    userEvent.click(showInternshipButton);

    expect(history.push).toHaveBeenCalledWith('/opportunities/6/virtual-internship/123');
  });

  it('opens recommend opportunity modal after clicking a button', async () => {
    renderComponentCardWithProvider([myOpportunityDetails]);

    const recommendOpportunityButton = await screen.findByRole('button', { name: 'Recommend' });
    userEvent.click(recommendOpportunityButton);

    expect(await screen.findByRole('dialog')).toBeInTheDocument();
    expect(screen.getByRole('dialog')).toHaveTextContent('Recommend an Opportunity');
  });
});
