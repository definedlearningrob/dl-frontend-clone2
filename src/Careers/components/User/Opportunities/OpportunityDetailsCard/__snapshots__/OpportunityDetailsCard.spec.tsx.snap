// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpportunityDetailsCard should render correctly 1`] = `
<div>
  <div
    class="card flex gap-base p-base xxxl:p-md"
    data-testid="opportunity-summary-card"
  >
    <div
      class="shrink-0 grow-0 basis-[min-content] min-w-[180px] xxxl:min-w-[240px] max-h-[180px] xxxl:max-h-[240px] flex justify-center"
    >
      <img
        class="object-cover rounded-sm max-w-full max-h-full m-auto"
        src="http://localstack.lvh.me:4566/dev-bucket/images/opportunities/tree=1"
      />
    </div>
    <div
      class="flex flex-col grow gap-sm"
    >
      <div>
        <h3
          class="text-sm mb-xs xxxl:text-lg xxxl:mb-sm"
        >
          Virtula Intern Ship 
        </h3>
        <div
          class="flex flex-col gap-xxs"
        >
          <div
            class="flex items-center gap-xxs xxxl:gap-xs text-font-secondary leading-lg whitespace-pre-wrap text-xxs xxxl:text-sm"
          >
            <div
              class="flex"
            >
              <div
                class="flex align-center justify-center grow-0 shrink-0"
                data-testid="icon"
              >
                <div
                  class="h-sm w-sm"
                />
              </div>
            </div>
            <span
              class="font-medium"
            >
              Type:
            </span>
            Virtual Internship
          </div>
          <div
            class="flex items-center gap-xxs xxxl:gap-xs text-font-secondary leading-lg whitespace-pre-wrap text-xxs xxxl:text-sm"
          >
            <div
              class="flex"
            >
              <div
                class="flex align-center justify-center grow-0 shrink-0"
                data-testid="icon"
              >
                <div
                  class="h-sm w-sm"
                />
              </div>
            </div>
            <span
              class="font-medium"
            >
              Pathway:
            </span>
            <span
              data-state="closed"
            >
              <span
                class="line-clamp-1 cursor-pointer"
              >
                A/V Technology & Film
              </span>
            </span>
          </div>
        </div>
      </div>
      <div>
        <div
          class="flex gap-xs xxxl:gap-sm"
        >
          <a
            href="/opportunities/6/virtual-internship/123"
          >
            <button
              class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] -primary bg-primary-500 hover:bg-primary-600 outline-primary-500 border-primary-500 py-xs px-sm !text-xs leading-[14px] !gap-xxs border text-white"
              data-testid="button"
              type="button"
            >
              <span
                class="items-center flex justify-center gap-xs"
              >
                Show internship
              </span>
            </button>
          </a>
          <button
            class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] -primary bg-primary-500 hover:bg-primary-600 outline-primary-500 border-primary-500 py-xs px-sm !text-xs leading-[14px] !gap-xxs border text-white"
            data-testid="button"
            type="button"
          >
            <span
              class="items-center flex justify-center gap-xs"
            >
              Recommend
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
